### PR TITLE
Update exception handling

### PR DESCRIPTION
**Pull request by Laurgrin**

### CODE REVIEW

File: src/Listener/RestExceptionListener.php [hunk 1]
```diff
@@ -55,7 +56,7 @@ public function onKernelException($event)
             return;
         }
 
-        $exception = $event->getException();
+        $exception = $event->getThrowable();
```
Comment (mSprunskas) [inline src/Listener/RestExceptionListener.php:59]: No such method in \Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent of symfony 3.4, see https://github.com/symfony/symfony/blob/3.4/src/Symfony/Component/HttpKernel/Event/GetResponseForExceptionEvent.php

**Review by mSprunskas — CHANGES_REQUESTED** [review]
Changes are backwards incompatible

File: src/Listener/RestExceptionListener.php (hunk 1 above)
Comment (Laurgrin) [reply src/Listener/RestExceptionListener.php]: Right, added BC support for a version that stopped it's support on 2021-11-30. :+1:

File: src/Listener/RestExceptionListener.php [hunk 2]
```diff
@@ -55,7 +56,16 @@ public function onKernelException($event)
             return;
         }
 
-        $exception = $event->getException();
+        if (method_exists($event, 'getException')) {
+            $exception = $event->getException();
+            trigger_deprecation(
```
Comment (mSprunskas) [inline src/Listener/RestExceptionListener.php:61]: BC support should not trigger deprecation notice

**Review by mSprunskas — CHANGES_REQUESTED** [review]

**Review by mSprunskas — APPROVED** [review]

**Pull request merged**